vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -30,7 +30,10 @@
 #include "base/strings/string_split.h"
 #include "base/time/time.h"
 #include "base/guid.h"
+
+#if defined(OS_ANDROID)
```

Comment:
sysinfo is not android specific so should be included for all platforms

Reply:
it's used only for android for this particular case (detecting system uptime)

Reply:
But as the code is not called and we need ways of testing this without building for all platforms can you see my example as this is why we use the ismobile and discussed is mobile and not preprocessor

Reply:
I still don't understand the reason of detecting OS during run-time. As you said- this code is only applicable for Android.  Please explain.

Reply:
As when developing features we test mobile business logic from desktop builds by changing what client info returns as otherwise we have to build across multiple platforms at the same time. This is consistent with how the libraries were first build due to being separate too

Reply:
don't agree

Reply:
I'm not sure as well why we should bring a code that is not used on desktop to a desktop. Do you have an idea @tmancey how would you emulate Android ads limit on desktop tests?